File: editaccount.py
import time
from selenium import webdriver
from selenium.webdriver.common.by import By
import pytest
from selenium.webdriver.common.alert import Alert
from selenium.common.exceptions import NoSuchElementException, WebDriverException
@pytest.fixture
def driver():
    driver = webdriver.Chrome()
    driver.maximize_window()
    yield driver
    driver.quit()
    
# Edit account
def test_edit_account_success(driver):
    try:
        # Login
        driver.get("http://localhost/ProjectWeb/login.html")
        time.sleep(2)
        driver.find_element(By.ID, "username").send_keys("abc@gmail.com")
        driver.find_element(By.ID, "password").send_keys("12345678")
        driver.find_element(By.XPATH, "//*[@id='login-form']/button").click()
        time.sleep(2)
        
        # Điều hướng đến trang user
        driver.get("http://localhost/ProjectWeb/userhtml.php")
        time.sleep(2)
        
        # Click vào nút sửa thông tin người dùng
        driver.find_element(By.XPATH, "/html/body/div[4]/table/tbody/tr[3]/td[4]/button[1]").click()
        
        # Sửa thông tin người dùng
        driver.find_element(By.ID, "fullname").clear()
        driver.find_element(By.ID, "fullname").send_keys("taun")
        
        # Click lưu thay đổi
        driver.find_element(By.XPATH, "/html/body/div[4]/table/tbody/tr[5]/td[4]/button[1]").click()
        
        # Nếu mọi thứ đều thành công, in thông báo
        print("Edit account successful")
        
    except (NoSuchElementException, WebDriverException) as e:
        # Nếu có lỗi, in thông báo lỗi
        print(f"An error occurred: {e}")

def test_edit_account_blank_fields(driver):#Bug 
    try:
        #Login 
        driver.get("http://localhost/ProjectWeb/login.html")
        time.sleep(2)
        driver.find_element(By.ID, "username").send_keys("abc@gmail.com")
        driver.find_element(By.ID, "password").send_keys("12345678")
        driver.find_element(By.XPATH, "//*[@id='login-form']/button").click()
        time.sleep(2)
        
        driver.get("http://localhost/ProjectWeb/userhtml.php")
        time.sleep(2)
        driver.find_element(By.XPATH, "/html/body/div[4]/table/tbody/tr[3]/td[4]/button[1]").click()
        # Sửa thông tin người dùng
        driver.find_element(By.ID, "fullname").clear()
        
        # Click lưu thay đổi
        driver.find_element(By.XPATH, "/html/body/div[4]/table/tbody/tr[5]/td[4]/button[1]").click()
        
        print('Edit account successful')

    except (NoSuchElementException, WebDriverException) as e:
        #Nếu có lỗi, in thông báo lỗi
        print(f"An error occurred: {e}")

# Khóa/Mở khóa tài khoản
def test_lock_account(driver):
    try :
        #Login 
        driver.get("http://localhost/ProjectWeb/login.html")
        time.sleep(2)
        driver.find_element(By.ID, "username").send_keys("abc@gmail.com")
        driver.find_element(By.ID, "password").send_keys("12345678")
        driver.find_element(By.XPATH, "//*[@id='login-form']/button").click()
        time.sleep(2)
        
        driver.get("http://localhost/ProjectWeb/userhtml.php")
        time.sleep(2)
        driver.find_element(By.XPATH, "/html/body/div[4]/table/tbody/tr[5]/td[5]/button").click()
        alert = Alert(driver)
        alert.accept()
        
        print("Account locked successfully")
        
        
    except (NoSuchElementException, WebDriverException) as e:
        print(f"An error occurred: {e}")

def test_unlock_account(driver):
    try:
        #Login 
        driver.get("http://localhost/ProjectWeb/login.html")
        time.sleep(2)
        driver.find_element(By.ID, "username").send_keys("abc@gmail.com")
        driver.find_element(By.ID, "password").send_keys("12345678")
        driver.find_element(By.XPATH, "//*[@id='login-form']/button").click()
        time.sleep(2)
        
        driver.get("http://localhost/ProjectWeb/userhtml.php")
        time.sleep(2)
        driver.find_element(By.XPATH, "/html/body/div[4]/table/tbody/tr[5]/td[5]/button").click()
        
        alert = Alert(driver)
        alert.accept()
            
        print("Account unlocked successfully")
        
        
    except (NoSuchElementException, WebDriverException) as e:
        print(f"An error occurred: {e}")

File: editaccount.php
<?php

use PHPUnit\Framework\TestCase;
use Facebook\WebDriver\Remote\RemoteWebDriver;
use Facebook\WebDriver\WebDriverBy;
use Facebook\WebDriver\WebDriverAlert;
use Facebook\WebDriver\Remote\DesiredCapabilities;
use Facebook\WebDriver\WebDriverExpectedCondition;

class AccountManagementTest extends TestCase
{
    /** @var RemoteWebDriver */
    private $driver;

    protected function setUp(): void
    {
        $host = 'http://localhost:4444/wd/hub'; // URL of the Selenium server
        $capabilities = DesiredCapabilities::chrome();
        $this->driver = RemoteWebDriver::create($host, $capabilities);
        $this->driver->manage()->window()->maximize();
    }

    protected function tearDown(): void
    {
        $this->driver->quit();
    }

    // Helper function for logging in
    private function login(string $username, string $password): void
    {
        $this->driver->get('http://localhost/ProjectWeb/login.html');
        usleep(2000000); // 2 seconds

        $this->driver->findElement(WebDriverBy::id('username'))->sendKeys($username);
        $this->driver->findElement(WebDriverBy::id('password'))->sendKeys($password);
        $this->driver->findElement(WebDriverBy::xpath("//*[@id='login-form']/button"))->click();

        usleep(2000000); // 2 seconds
    }

    // Test case: Edit account successfully
    public function testEditAccountSuccess(): void
    {
        try {
            $this->login('abc@gmail.com', '12345678');

            $this->driver->get('http://localhost/ProjectWeb/userhtml.php');
            usleep(2000000);

            $editButton = $this->driver->findElement(WebDriverBy::xpath("/html/body/div[4]/table/tbody/tr[3]/td[4]/button[1]"));
            $editButton->click();

            $fullnameField = $this->driver->findElement(WebDriverBy::id('fullname'));
            $fullnameField->clear();
            $fullnameField->sendKeys('taun');

            $saveButton = $this->driver->findElement(WebDriverBy::xpath("/html/body/div[4]/table/tbody/tr[5]/td[4]/button[1]"));
            $saveButton->click();

            echo "Edit account successful\n";
        } catch (NoSuchElementException $e) {
            echo "Element not found during account edit: " . $e->getMessage() . "\n";
        } catch (WebDriverException $e) {
            echo "WebDriver error occurred: " . $e->getMessage() . "\n";
        }
    }

    // Test case: Edit account with blank fields (Bug simulation)
    public function testEditAccountBlankFields(): void
    {
        try {
            $this->login('abc@gmail.com', '12345678');

            $this->driver->get('http://localhost/ProjectWeb/userhtml.php');
            usleep(2000000);

            $editButton = $this->driver->findElement(WebDriverBy::xpath("/html/body/div[4]/table/tbody/tr[3]/td[4]/button[1]"));
            $editButton->click();

            $fullnameField = $this->driver->findElement(WebDriverBy::id('fullname'));
            $fullnameField->clear();

            $saveButton = $this->driver->findElement(WebDriverBy::xpath("/html/body/div[4]/table/tbody/tr[5]/td[4]/button[1]"));
            $saveButton->click();

            echo "Edit account successful\n";
        } catch (NoSuchElementException $e) {
            echo "Element not found during account edit with blank fields: " . $e->getMessage() . "\n";
        } catch (WebDriverException $e) {
            echo "WebDriver error occurred: " . $e->getMessage() . "\n";
        }
    }

    // Test case: Lock account
    public function testLockAccount(): void
    {
        try {
            $this->login('abc@gmail.com', '12345678');

            $this->driver->get('http://localhost/ProjectWeb/userhtml.php');
            usleep(2000000);

            $lockButton = $this->driver->findElement(WebDriverBy::xpath("/html/body/div[4]/table/tbody/tr[5]/td[5]/button"));
            $lockButton->click();

            $alert = $this->driver->switchTo()->alert();
            $alert->accept();

            echo "Account locked successfully\n";
        } catch (NoSuchElementException $e) {
            echo "Element not found during account lock: " . $e->getMessage() . "\n";
        } catch (WebDriverException $e) {
            echo "WebDriver error occurred: " . $e->getMessage() . "\n";
        }
    }

    // Test case: Unlock account
    public function testUnlockAccount(): void
    {
        try {
            $this->login('abc@gmail.com', '12345678');

            $this->driver->get('http://localhost/ProjectWeb/userhtml.php');
            usleep(2000000);

            $unlockButton = $this->driver->findElement(WebDriverBy::xpath("/html/body/div[4]/table/tbody/tr[5]/td[5]/button"));
            $unlockButton->click();

            $alert = $this->driver->switchTo()->alert();
            $alert->accept();

            echo "Account unlocked successfully\n";
        } catch (NoSuchElementException $e) {
            echo "Element not found during account unlock: " . $e->getMessage() . "\n";
        } catch (WebDriverException $e) {
            echo "WebDriver error occurred: " . $e->getMessage() . "\n";
        }
    }
}
